### PR TITLE
handbrake: 1.3.1 -> 1.3.2

### PR DESCRIPTION
**Pull request by Anton-Latukha**
###### Motivation for this change

Starting to use GitHub CDN.
Every time there is any HandBrake release - it gets published on their website after a timeout of arbitrary days. And I want to be able to respond to notifications of release right away, and not postpone, stagger, keep checking wait website tarball and remember that I still need to update.

Also update to 1.3.2.

###### Things done

- [x] Tested using sandboxing ([nix.useSandbox](https://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `sandbox` in [`nix.conf`](https://nixos.org/nix/manual/#sec-conf-file) on non-NixOS linux)
- Built on platform(s)
   - [x] NixOS
   - [ ] macOS
   - [ ] other Linux distributions
- [x] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [ ] Tested compilation of all pkgs that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review wip"`
- [x] Tested execution of all binary files (usually in `./result/bin/`)
- [x] Determined the impact on package closure size (by running `nix path-info -S` before and after)
- [x] Ensured that relevant documentation is up to date
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

### CODE REVIEW

**Comment by peterhoeg** [comment]
What happens when you build from the checkout as opposed to the release tarball?

**Comment by peterhoeg** [comment]
and add `autoreconfHook` to `nativeBuildInputs`.

**Comment by doronbehar** [comment]
Also, some parts of the `preConfigure` are not needed since:

https://github.com/HandBrake/HandBrake/issues/2690

https://github.com/HandBrake/HandBrake/pull/2712

**Comment by Anton-Latukha** [comment]
@peterhoeg: 
> and add `autoreconfHook` to `nativeBuildInputs`.

Results:
```
source root is HandBrake-1.3.2
setting SOURCE_DATE_EPOCH to timestamp 1588534233 of file HandBrake-1.3.2/version.txt
patching sources
autoreconfPhase
autoreconf: 'configure.ac' or 'configure.in' is required
builder for '/nix/store/dvg6dlm5pwrxq8pfbljjqjfky6v5hsra-handbrake-1.3.2.drv' failed with exit code 1
```

**Comment by Anton-Latukha** [comment]
@peterhoeg:
> What happens when you build from the checkout as opposed to the release tarball?

### Git checkout

I was discussing this issue with them upstream a couple of years ago. They have invented their-own versioning inside a project:
```
nix-shell . -A handbrake
unpackPhase
cd source
patchPhase
scripts/repo-info.sh    # source: https://github.com/HandBrake/HandBrake/blob/8e2a08267705f09e92f06289820ad6ebbfc9040e/scripts/repo-info.sh
```

Output:

> URL=
> HASH=ccb2e7234e01a2a07f537512045f7511583deaf9
> SHORTHASH=ccb2e7234e0
> REV=224188
> BRANCH=handbrake-1.3.2
> REMOTE=
> DATE=2020-05-05 12:44:02 +0300

And this returned information results into `probe: repo info...(fail) code 1`

Results in build log:
```
source root is source
patching sources
configuring
...
probe: repo info...(fail) code 1
probe: version.txt...(fail)
ERROR: HandBrake is missing version information it needs to build properly.
Clone the official git repository at https://github.com/HandBrake/HandBrake.git
or download an official source archive from https://handbrake.fr
; configure stop.
compute: project data...
builder for '/nix/store/vwispj413x66pgh419awlgd6z9vj3nnv-handbrake-1.3.2.drv' failed with exit code 1
```

I got no description on what this situation is about, threads I've read resolve things replying "use website tarball".

### Tarball package

Results in:
```
scripts/repo-info.sh
```

This also results into `probe: repo info...(fail) code 1`, but now of the other kind

> fatal: not a git repository (or any of the parent directories): .git
> fatal: not a git repository (or any of the parent directories): .git
> Not a valid repository.

And that situation "works" in the tarball:

```
probe: repo info...(fail) code 1
probe: version.txt...(pass)
...
```

If to give the long answer.

As I understand that there does not exist the case to succesfully go there proselytize and rip-out their internal distribution & versioning - I do not worry about this too much.

**Comment by Anton-Latukha** [comment]
> Also, some parts of the `preConfigure` are not needed since:
> 
> [HandBrake/HandBrake#2690](https://github.com/HandBrake/HandBrake/issues/2690)
> 
> [HandBrake/HandBrake#2712](https://github.com/HandBrake/HandBrake/pull/2712)

@doronbehar 

Seems done.

Also seems that upstream fixes around `libxml2` not fixed some of the `libxml2` cases we receiving..

File: pkgs/applications/video/handbrake/default.nix
Comment (doronbehar) [inline pkgs/applications/video/handbrake/default.nix:12]: Tested:

```suggestion
  python3, pkgconfig, autoconf, automake, libtool, m4, lzma,
```

File: pkgs/applications/video/handbrake/default.nix
Comment (doronbehar) [inline pkgs/applications/video/handbrake/default.nix:85]: Just pointing out that I confirm this is still needed.

**Comment by doronbehar** [comment]
BTW, I'm getting this when running it:

```
** Message: 14:08:09.629: Couldn't initialize gstreamer. Disabling live preview.
```

I think you can fix it with something similar to:

https://github.com/Anton-Latukha/nixpkgs/blob/f9a7d6c4ed32bb78d0df7d25dd1a3b72f658dbe4/pkgs/applications/audio/transcribe/default.nix#L46-L47

While some gst-plugins are located in the build inputs so `wrapGAppsHook` will pick them up.

But it's also possible that `gappsWrapperArgs+=(` is not needed...

**Comment by Anton-Latukha** [comment]
> BTW, I'm getting this when running it:
> 
> ```
> ** Message: 14:08:09.629: Couldn't initialize gstreamer. Disabling live preview.
> ```

I specifically tested that live preview working. Since there is already wrapper in place, I would just add configuration for the wrapper.

**Comment by doronbehar** [comment]
> @doronbehar
> I specifically tested that live preview works.
> And with https://github.com/Anton-Latukha/nixpkgs/blob/f9a7d6c4ed32bb78d0df7d25dd1a3b72f658dbe4/pkgs/applications/audio/transcribe/default.nix#L46-L47 ghb still throws that same warning. And just as before - in ghb live preview still works despite of that warning.

OK I guess then it's not that bad (I'm not using Handbrake actively, I only tested it once in the past). If someone will notice some missing functionality that may be related to this message, he'll put the effort to debug this.

**Review by doronbehar — APPROVED** [review]

**Comment by Anton-Latukha** [comment]
@doronbehar 

It makes in `Preview` the `Live Preview` button appear.
Like here: https://handbrake.fr/docs/en/latest/workflow/preview-settings.html

I also do not use HandBrake mutch these days, Preview is needed, I almost never run Live Preview.

Wrapper fix did not helped.
I may dig into it, construct `--prefix LD_LIBRARY_PATH : "${libPath}"` and assemble its `libPath` variable or make deeper assessment.

**Comment by Anton-Latukha** [comment]
I do not understand why if one maintainer changes to the package, and already went through the review process - why other same-level maintained should "mandatory" review it.

The Mertens for example probably or too busy or do not care for it long time ago.

The review is done.

**Comment by doronbehar** [comment]
Yea this is how things are in this repo unfortunately. Sometimes a maintainer with a merge bit goes through the reviewed pulls sorted from old to new and they may encounter this PR at some point, but don't hold your breath for this to happen.

**Comment by wmertens** [comment]
Sorry, I missed that it was ok to merge. Indeed, I don't use handbrake any more, so feel free to sign yourself as a maintainer

**Comment by Anton-Latukha** [comment]
@wmertens I am also a maintainer of this package for 2-3 years already.

We chatted with you back in the day about the package.

I am talking that I am a maintainer, process requires review from the higher-order maintainer, like `doronbehar`.

But I am maintainer that passes review.
I am just puzzled why "process" requires review from same-level package maintainers.

People tend to not remove themself from maintainer fields and not care and not show up.
And so - there are packages that have 5-6 "maintainers" but no one shows up to maintain it. So why that package should have all 5-6 maintainers sign-off between each other every time? It is not like this is Linux Kernel development, and as I know they not do such multicross signoff all the time.

**Comment by Anton-Latukha** [comment]
@doronbehar

Thanks to you a lot. You are a great guy.

How is auto update bot there.

I see that it still not pipelined to autotrack and automerge

Why? It seems really strange.

I think most people woul've preferred for minor updates to go without bothering them.

**Comment by doronbehar** [comment]
> Thanks to you a lot. You are a great guy.

:green_heart: 

> It is not like this is Linux Kernel development, and as I know they do multistage reviews, but do not require such multicross peer review signoff all the time, because nothing would get done then.

> How is auto update bot there.
> I see that it still not pipelined to autotrack and automerge
> Why? It seems really strange.
> I think most people, and you, would've preferred for minor updates to go without bothering them.

Your phrasing is not that good but I think I understand you. See:

- https://github.com/NixOS/rfcs/pull/50
- https://discourse.nixos.org/t/new-rfc-50-merge-bot-for-maintainers/3795
- https://discourse.nixos.org/t/steps-towards-even-more-pr-automation/5634/13
- https://discourse.nixos.org/t/suggestion-to-help-maintainers-with-nixpkgs-prs-bloat/4794/20
- https://github.com/NixOS/rfcs/pull/51
- https://github.com/NixOS/rfcs/pull/30

**Pull request merged**
